sqlglot/dialects/hive.py

```diff
@@ -599,15 +599,52 @@ class Generator(generator.Generator):
 
         PROPERTIES_LOCATION = {
             **generator.Generator.PROPERTIES_LOCATION,
+            exp.DistributedByHash: exp.Properties.Location.POST_SCHEMA,
+            exp.DistributedByRandom: exp.Properties.Location.UNSUPPORTED,
+            exp.DuplicateKey: exp.Properties.Location.UNSUPPORTED,
+            exp.EngineProperty: exp.Properties.Location.UNSUPPORTED,
             exp.FileFormatProperty: exp.Properties.Location.POST_SCHEMA,
             exp.PartitionedByProperty: exp.Properties.Location.POST_SCHEMA,
+            exp.PrimaryKey: exp.Properties.Location.UNSUPPORTED,
             exp.VolatileProperty: exp.Properties.Location.UNSUPPORTED,
             exp.WithDataProperty: exp.Properties.Location.UNSUPPORTED,
         }
 
+        # https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL
+        def distributedbyhash_sql(self, expression: exp.DistributedByHash) -> str:
+            expressions = self.expressions(expression, key="expressions", flat=True)
+            sorted_by = self.expressions(expression, key="sorted_by", flat=True)
+            sorted_by = f" SORTED BY ({sorted_by})" if sorted_by else ""
+            buckets = "1"
+            if expression.auto_bucket:
+                self.unsupported(
+                    "DistributedByHash without buckets, Spark requires a number of buckets"
+                )
+            else:
+                buckets = self.sql(expression, "buckets")
+            return f"CLUSTERED BY ({expressions}){sorted_by} INTO {buckets} BUCKETS"
+
+        def doris_or_sr_type_to_hive(self, expression: exp.DataType) -> str:
+            if expression.is_type(
+                exp.DataType.Type.SMALLINT,
+                exp.DataType.Type.INT,
+                exp.DataType.Type.BIGINT,
+                exp.DataType.Type.FLOAT,
+                exp.DataType.Type.DOUBLE,
+            ):
+                # To ignore the DataTypeParam. e.g. INT(11) -> INT
+                return expression.this.name
+            return super().datatype_sql(expression)
+
+        def property_name(self, expression: exp.Property, string_key: bool = False) -> str:
+            return super().property_name(expression, True)
+
         def unnest_sql(self, expression: exp.Unnest) -> str:
             return rename_func("EXPLODE")(self, expression)
 
+        def with_properties(self, properties: exp.Properties) -> str:
+            return self.properties(properties, prefix=self.seg("TBLPROPERTIES", sep=""))
```

Comment:
Why did this change?

Reply:
I saw that hive has a definition:
WITH_PROPERTIES_PREFIX = "TBLPROPERTIES"
But after transpile, it seems that the expected TBLPROPERTIES is not obtained, as follows:
```
PROPERTIES (
'replication_num'='1',
```
> read='starrocks', dialect='hive'

Overwride the with_properties may not be good 🤔